koschei/backend/services/resolver.py

```diff
@@ -369,7 +369,7 @@ def persist_resolution_output(self, chunk):
         self.db.commit_no_expire()
 
         # emit fedmsg (if enabled)
-        if state_changes:
+        if state_changes and not repo_fixed:
```

Comment:
- by "repo fixed" or "broken" I mean transition from collection state "unresolved" to "ok" or vice versa, respectively, i.e. "repo fixed" means collection was previously unresolved, but it was resolved in current resolver run
- when "repo is broken", all packages that were ok or failing suddenly become unresolved, yet we intentionally suppress sending fedmsg for package state changes
- likewise, when "repo is fixed" we should suppress sending fedmsg; this is what this check is for